test/writer.asciidoc

```diff
@@ -466,7 +466,11 @@ Ellipses…and…and….
 * latexmath:[$223$]
 * latexmath:[$p$]-Tree
 * Here’s some display math:
-latexmath:[\[\frac{d}{dx}f(x)=\lim_{h\to 0}\frac{f(x+h)-f(x)}{h}\]]
+
```

Comment:
After experimenting, I can confirm that the math in the list item doesn't work. The blank line causes the list to be closed, then the display math occurs, then a new list.  In order to get the right result you need
```
* Here's some display math:
+
[latexmath]
++++
etc
```
But note: we *only* want the single `+` connector if we're in a list item.

Reply:
Also, asciidoctor seems to choke on the empty list item.
We might fix that somehow (maybe inserting  a nonbreaking space?), but that doesn't need to go in this PR.

Reply:
I think you can check to see if you're in a list by looking at orderedListLevel and bulletListLevel in WriterState.
If they're both == 1 you're not in a list.